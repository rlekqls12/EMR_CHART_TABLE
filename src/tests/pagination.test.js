import React from 'react';
import { mount, shallow } from 'enzyme';
import Pagination from 'src/components/pagination';

describe('<Pagination />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders pagination', () => {
    let nowPage = 1;
    const onPageChange = (now) => nowPage = now;
    const wrapper = mount(<Pagination initialPage={1} maxPage={2} showPageList={2} onPageChange={onPageChange} />);

    const divElement = wrapper.find('div');
    expect(divElement.exists()).toBeTruthy();

    const prevPageElement = wrapper.findWhere(node =>
      node.type() === 'button' && node.text() === '◀'
    );
    expect(prevPageElement.exists()).toBeTruthy();

    const nextPageElement = wrapper.findWhere(node =>
      node.type() === 'button' && node.text() === '▶'
    );
    expect(nextPageElement.exists()).toBeTruthy();

    prevPageElement.simulate('click');
    expect(nowPage).toBe(1);

    nextPageElement.simulate('click');
    expect(nowPage).toBe(2);

    nextPageElement.simulate('click');
    expect(nowPage).toBe(2);

    prevPageElement.simulate('click');
    expect(nowPage).toBe(1);

    const page2Element = wrapper.findWhere(node =>
      node.type() === 'span' && node.text() === '2'
    );
    expect(page2Element.exists()).toBeTruthy();
    page2Element.simulate('click');
    expect(nowPage).toBe(2);
  });
});