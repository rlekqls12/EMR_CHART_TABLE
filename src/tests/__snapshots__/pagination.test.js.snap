// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches snapshot 1`] = `ReactWrapper {}`;
